components/IdentityProvider.tsx

```diff
@@ -29,14 +30,23 @@ const styles = {
   },
 } satisfies StyleConfig;
 
+function IdentityProviderContainer() {
+  // Wait for the Stytch user object to be loaded by the SDK to prevent flicker on first load
+  const { user } = useStytchUser();
+  if (!user) {
+    return null;
+  }
+  return <BaseIdentityProvider styles={styles} />;
+}
+
 // The IdentityProvider is the only part of this demo that uses the Stytch B2C UI components
 // During local development, it reads from the stytch_session cookie stored by the backend servies
 // During remote deployment, CNAMEs are utilized to share the cookie information between your
 // backend API (e.g. customer.com) and the Stytch backend (e.g. auth.customer.com)
 export function IdentityProvider() {
   return (
     <StytchProvider stytch={client}>
-      <BaseIdentityProvider styles={styles} />
+      <IdentityProviderContainer />
     </StytchProvider>
   );
 }
```

Comment:
feel free to defer to a future PR, but the naming of the components in this file is kind of confusing at this point: `IdentityProvider` contains `IdentityProviderContainer` contains `BaseIdentityProvider`

i would recommend moving the user check into the `IdentityProvider` component and getting rid of the `[...]Container`, _or_ maybe just renaming `IdentityProviderContainer` to `BaseIdentityProviderContainer`(though i am personally not a huge fan of having too many wrappers).


```suggestion
// The IdentityProvider is the only part of this demo that uses the Stytch B2C UI components
// During local development, it reads from the stytch_session cookie stored by the backend servies
// During remote deployment, CNAMEs are utilized to share the cookie information between your
// backend API (e.g. customer.com) and the Stytch backend (e.g. auth.customer.com)
export function IdentityProvider() {
  // Wait for the Stytch user object to be loaded by the SDK to prevent flicker on first load
  const { user } = useStytchUser();
  return (
    <StytchProvider stytch={client}>
      { user ? <BaseIdentityProvider styles={styles} /> : null }
    </StytchProvider>
  );
}
```

Reply:
Sadly we need to call `useStytchUser` within the confines of the StytchProvider so we end up in container heck